src/praisonai/praisonai/persistence/conversation/sync_sqlite.py

```
@@ -0,0 +1,329 @@
+            conn.close()
+    
+    def add_message(self, message: ConversationMessage) -> ConversationMessage:
+        """Add a message to the conversation."""
+        with self._lock:
+            conn = self._get_connection()
+            try:
+                table = f"{self.table_prefix}messages"
+                conn.execute(f"""
+                    INSERT INTO {table} (id, session_id, role, content, tool_calls, tool_call_id, metadata, created_at)
+                    VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+                """, (
```

Ruff (0.15.14)

[error] Possible SQL injection vector through string-based query construction

(S608)